doc/source/tools/gmsh/gmsh.rst

```diff
@@ -0,0 +1,371 @@
+4. Click on the right line for no condition.
+
+By reloading the script, you will see those four lines of code appear:
+
+.. code-block::
+	
+	// Physical Curve(<id>) = {<id of curve element>, ...}
+	Physical Curve(1) = {7};
+	Physical Curve(2) = {6, 9};
+	Physical Curve(3) = {5};
+	Physical Curve(4) = {8};
+	
+.. important::
+	The ``bc #`` in the ``.prm`` files (:doc:`../../parameters/cfd/boundary_conditions_cfd`) must be the same as the ``id`` specified for the physical group.
+
+Then, add a ``Physical Surface`` for a 2D geometry, or a ``Physical Volume`` for a 3D geometry. 
+
+.. warning::
+	All cells must be in a Physical group.
+	
+	While not adding a ``Physical Surface`` or ``Volume`` would not prevent GMSH from building the mesh, it would result in an error when the mesh is loaded by deal.II. This is often a source of error.
+
+.. code-block::
+	
+	// Physical Surface(id) = {<id of surface element>, ...}
+	Physical Surface(1) = {1};
+	
+Then, define the boundary conditions accordingly in the parameter file :
+
+.. code-block:: text
+
+	subsection boundary conditions
+	  set number                  = 4
+	  subsection bc 0
+		set id                = 1
+		set type              = function
+		subsection u
+		    set Function expression = 1
+		end
+		subsection v
+            	    set Function expression = 0
+        	end
+	  end
+	  subsection bc 1
+	  	set id                = 2
+		set type              = slip
+	  end
+	  subsection bc 2
+	  	set id                = 3
+		set type              = noslip
+	  end
+	  subsection bc 3
+	  	set id                = 4
+	  	set type              = none
+	end
+
+.. _mesh:
+
+---------------------------
+Mesh
+---------------------------
+
+""""""""""""""""""""""""""
+Unstructured
+""""""""""""""""""""""""""
+Basic:
+
+1. (optional) ``Tools > Options > Mesh`` and ``General`` panel, check ``Recombine all triangular meshes``: generate a quad mesh.
+2. (optional) In the same panel, change ``Min/Max element size`` to have smaller/bigger elements, therefor a finer/coarser mesh.
+3. ``Left pannel: Modules > Mesh > 2D`` or ``3D``: create the mesh
+4. ``Tools > Statistics``: check that the mesh is generated appropriately (by default, triangles for 2D and hexahedra for 3D)
+5. (optional) ``Left pannel: Modules > Mesh > Refine by splitting``: refine the mesh (beware, it takes more and more time for each refinement)
+6. ``Left pannel: Modules > Mesh > Save``: save the mesh in a ``.msh`` file, to be used in Lethe (see :doc:`../../parameters/cfd/mesh`)
+
+By following all the previous steps, the mesh generated looks like bellow.
+
+.. image:: images/unstructured.png
+    :alt: 2D mesh with quads
+    :align: center
+    :name: unstructured mesh
+    
+Attractors can also be used to refine the mesh towards specific edges or surfaces. In this example, attractors could be interesting if the mesh needs to be finer around the sphere. Attractors can only be added by code with the ``Field`` attribut.
```

Comment:
Should there be a subsection title called Attractors?

Reply:
Agreed. Or maybe local mesh refinement? Because this is what Attractors achieve.

Reply:
Good idea!